x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -1212,3 +1177,94 @@ func keys(m map[string]interface{}) []string {
 	}
 	return keys
 }
+
+func datasetNamespaceFromInputNode(inputNode Node) string {
+	const defaultNamespace = "default"
+
+	if namespaceNode, found := inputNode.Find("dataset.namespace"); found {
```

Comment:
Looks like we have very similar code in several places. Two things I'm wondering: Could go-ucfg to this for you and if not, could we extract it into a method?